sky/backends/backend_utils.py

```diff
@@ -730,10 +730,16 @@ def write_cluster_config(
         - 'ray'
         - 'tpu-create-script' (if TPU is requested)
         - 'tpu-delete-script' (if TPU is requested)
+    Raises:
+        ResourceUnavailableError: if the region/zones requested does not appear
+            in the catalog.
     """
     # task.best_resources may not be equal to to_provision if the user
     # is running a job with less resources than the cluster has.
     cloud = to_provision.cloud
+    # This can raise a ResourceUnavailableError, when the region/zones requested
+    # does not appear in the catalog. It can be triggered when the user changed
+    # the catalog file, while there is a cluster in the removed region/zone.
```

Comment:
That is a good point! I agree that we may probably have a different exception for the resources not found in the catalog. Also, it would be better if we can check the issue before this function, as this function name does not indicates any checks, but just translating the requirement into the ray yaml. Added a TODO for it.

Reply:
Totally agreed that this func name does not suggest it'd do anything that can trigger exceptions. The TODO looks good to me.